components/Snackbar/src/private/MDCSnackbarOverlayView.m

```diff
@@ -257,85 +257,79 @@ - (void)setSnackbarView:(MDCSnackbarMessageView *)snackbarView {
       // Pin the snackbar to the bottom of the screen.
       [snackbarView setTranslatesAutoresizingMaskIntoConstraints:NO];
 
-      switch (self.traitCollection.horizontalSizeClass) {
-        case UIUserInterfaceSizeClassUnspecified:
-        case UIUserInterfaceSizeClassCompact:
-
+      BOOL isRegularWidth = self.traitCollection.horizontalSizeClass == UIUserInterfaceSizeClassRegular;
+      BOOL isRegularHeight = self.traitCollection.verticalSizeClass == UIUserInterfaceSizeClassRegular;
+      if (isRegularWidth && isRegularHeight) {
+        snackbarView.centerConstraint  = [NSLayoutConstraint constraintWithItem:snackbarView
+                                                                      attribute:NSLayoutAttributeCenterX
+                                                                      relatedBy:NSLayoutRelationEqual
+                                                                         toItem:container
+                                                                      attribute:NSLayoutAttributeCenterX
+                                                                     multiplier:1.0
+                                                                       constant:0];
+        snackbarView.centerConstraint.active = self.alignment == MDCSnackbarAlignmentCenter;
+
+        snackbarView.leadingConstraint  = [NSLayoutConstraint constraintWithItem:snackbarView
+                                                                       attribute:NSLayoutAttributeLeading
+                                                                       relatedBy:NSLayoutRelationEqual
+                                                                          toItem:container
+                                                                       attribute:NSLayoutAttributeLeading
+                                                                      multiplier:1.0
+                                                                        constant:sideMargin];
```

Comment:
I like the use of Regular/Regular here, and I think it aligns with the Material Guidelines. Looks like sideMargin is dependent on device idiom (iPad/iPhone). Should we migrate that code to size classes as well? I'm concerned that a split-screen iPad in a Regular(H)/Compact(W) environment will have the wrong margins (24 instead of 8).